docs(dialog)integration with other components (#2025)
telerik/blazor-docs: components/dialog/integration.md

---
title: Integration
page_title: Dialog Integration

> description: Discover the Blazor Dialog Integration and explore the examples.

Reviewer: * The description is below the minimum required length (100 characters - min, 150 characters - optimal).
* It's vague (you get no idea what the content would be without reading the article because "Blazor Dialog Integration" is too fuzzy).
* See the [Metadata ](https://docs.telerik.com/style-guide/metadata#creating-effective-meta-descriptions)article for tips on how to write effective SEO descriptions.

```suggestion
description: Learn how to integrate Blazor components like the Checkbox and the Filter into the Blazor Dialog and explore the practical sample code.
```

slug: dialog-integration
tags: telerik,blazor,dialog,integration
published: True
position: 9
---

> # Blazor Dialog Integration

Reviewer: If I look only at the title, I would think that the article's topic is how the Dialog integrates _into_ other components. Try being more specific.
```suggestion
# Integrating Components into the Blazor Dialog
```

> You can use other components inside the `DialogContent`. In this case, the two-way binding for the parameters of these components may not work as expected. The Dialog needs to [`Refresh`]({%slug dialog-overview%}#dialog-reference-and-methods) to reflect UI changes.

Reviewer: * A common practice for the opening paragraph is to start with "Why", e.g. name the use cases. This allows you to set the context and the article reads much easier.
* Show the binding information in a note as it is a limitation that must be considered. The note allows you to highlight this requirement.
```suggestion
A common application requirement is to display other Blazor components in the Blazor Dialog. You can achieve this by placing them inside the `DialogContent`.

>The two-way binding for the parameters of the components nested in the `DialogContent` may not work as expected. The Dialog needs to [`Refresh`]({%slug dialog-overview%}#dialog-reference-and-methods) to reflect UI changes.
```

This article contains the following examples:

* [Checkbox integration in Dialog](#checkbox-in-a-dialog)
* [Filter integration in Dialog](#filter-in-a-dialog)

## Checkbox in a Dialog

> **To integrate the Checkbox in the Dialog, you can:**

Reviewer: Avoid using bold unless prescribed by the [guidelines ](https://docs.telerik.com/style-guide/element-formatting#guidelines)in the Style Guide.
```suggestion
To integrate the Checkbox in the Dialog:
```

1. Include the [Telerik Checkbox]({%slug checkbox-overview%}) as `DialogContent`.
1. Set the [`Value` parameter]({%slug checkbox-overview%}#checkbox-parameters) of the Checkbox via two-way binding.
1. Invoke the Dialog's `Refresh` method in the [`OnChange` event]({%slug checkbox-events%}#onchange) of the Checkbox.

> >caption Use Checkbox in Dialog.

Reviewer: ```suggestion
>caption Using Checkbox in Dialog
```

````CSHTML
@using Telerik.DataSource

<TelerikDialog @ref="DialogRef" Visible="true">
    <DialogContent>
        <TelerikCheckBox Id="MyCheckBox" @bind-Value="@IsSelected" OnChange="@OnCheckBoxValueChanged" />
        <label for="MyCheckBox">@(IsSelected ? "Selected" : "Not selected")</label>
    </DialogContent>
</TelerikDialog>

@code {
    private bool IsSelected { get; set; }

    private TelerikDialog DialogRef { get; set; }

    private void OnCheckBoxValueChanged()
    {
        DialogRef.Refresh();
    }
}
````

## Filter in a Dialog

> **To integrate the Filter in the Dialog, you can:**

Reviewer: ```suggestion
To integrate the Filter in the Dialog:
```

1. Include the [Telerik Filter]({%slug filter-overview%}) as `DialogContent`.
1. Set the [`Value` parameter]({%slug filter-overview%}#filter-parameters) of the Filter via one-way binding.
1. Invoke the Dialog's `Refresh` method in the [`ValueChanged` event]({%slug filter-events%}#valuechanged) of the Filter.
1. Update the `Value` parameter of the Filter manually in the `ValueChanged` event of the Filter.

> >caption Use Filter in Dialog.

Reviewer: ```suggestion
>caption Using Filter in Dialog
```
